cobbler/utils/filesystem_helpers.py

```diff
@@ -317,6 +336,29 @@ def mkdir(path: str, mode: int = 0o755) -> None:
             raise CX(f"Error creating {path}") from os_error
 
 
+def mkdirimage(path: pathlib.Path, image_location: str) -> None:
+    """
+    Create a directory in an image.
+
+    :param path: The path to create the directory at.
+    :param image_location: The location of the image.
+    """
+
+    path_parts = path.parts
+    cmd = ["mmd", "-i", image_location, path]
+    path.mkdir(parents=True, exist_ok=True)
+    try:
+        # Create all parent directories one by one
+        for parent_directory in range(1, len(path_parts) + 1):
+            cmd[-1] = "/".join(path_parts[:parent_directory])
+            logger.info('running: "%s"', cmd)
+            subprocess.run(cmd, check=True)
```

Comment:
Hmm, I don't really get this, maybe you can clarify this to me:

- The comment for the for loop mentions that the loop is creating all parent directories, but you are calling  `path.mkdir(parents=True, exist_ok=True)` which should create them. Am I missing something?
- Why are we calling `mmd` several times inside the loop, so one call for each one of the parent directories? Is this intended?

Maybe I'm missing something here, could you please clarify this better maybe also as code comments?

Thanks!

Reply:
> * The comment for the for loop mentions that the loop is creating all parent directories, but you are calling  `path.mkdir(parents=True, exist_ok=True)` which should create them. Am I missing something?

right `path.mkdir(parents=True, exist_ok=True)` it's not required

> * Why are we calling `mmd` several times inside the loop, so one call for each one of the parent directories? Is this intended?
> 
probably it's more clear now that I removed the wrong `path.mkdir` ...anyway `mmd` creates folder inside the image, but there's  anything similar to the argument `-p` of `mkdir`

> 
> Thanks!

Reply:
not sure why but removing it cause an issue in `pytest`

Reply:
I haven't succeed to create the entire path in the image recursively in one run with the `mmd` command.That's why I've recreated the logic here. There's not that much documentation on it, so this was the working solution at least. I've probably missed something.

Reply:
@meaksh